woltka/workflow.py

```diff
@@ -0,0 +1,590 @@
+        Newick tree file.
+    lineage_fp : str, optional
+        Lineage strings file.
+    ranktb_fp : str, optional
+        Rank table file.
+    map_fps : list of str, optional
+        Mapping files.
+    map_rank : bool, optional
+        Mapping filename is rank.
+
+    Returns
+    -------
+    dict, dict, dict, str
+        Taxonomic tree.
+        Rank dictionary.
+        Name dictionary.
+        Root identifier.
+    """
+    tree, rankd, named = {}, {}, {}
+    is_build = any([
+        names_fp, nodes_fp, newick_fp, lineage_fp, ranktb_fp, map_fps])
+    if is_build:
+        click.echo(f'Reading classification system...', nl=False)
+
+    # taxonomy names
+    if names_fp:
+        with readzip(names_fp) as f:
+            named = read_names(f)
+
+    # taxonomy nodes
+    if nodes_fp:
+        with readzip(nodes_fp) as f:
+            tree_, rankd_ = read_nodes(f)
+        update_dict(tree, tree_)
+        update_dict(rankd, rankd_)
+
+    # Newick-format tree
+    if newick_fp:
+        with readzip(newick_fp) as f:
+            update_dict(tree, read_newick(f))
+
+    # lineage strings file
+    if lineage_fp:
+        with readzip(lineage_fp) as f:
+            tree_, rankd_ = read_lineage(f)
+        update_dict(tree, tree_)
+        update_dict(rankd, rankd_)
+
+    # rank table file
+    if ranktb_fp:
+        with readzip(ranktb_fp) as f:
+            update_dict(tree, read_ranktb(f))
+
+    # plain mapping files
+    for fp in map_fps:
+        rank = path2stem(fp)  # filename stem as rank
+        with readzip(fp) as f:
+            map_ = read_map(f)
+        update_dict(tree, map_)
+        if map_rank:
+            update_dict(rankd, {k: rank for k in set(map_.values())})
+
+    # fill root
+    root = fill_root(tree)
+
+    if is_build:
+        click.echo(' Done.')
+        click.echo(f'Total classification units: {len(tree)}.')
+    return tree, rankd, named, root
+
+
+def reshape_rmap(rmap:    dict,
+                 deidx:   bool = None,
+                 demux:   bool = None,
+                 samples: list = None,
+                 files:   dict = None,
+                 fp:      str = None) -> dict:
+    """Reshape a read map.
+
+    Parameters
+    ----------
+    rmap : dict
+        Read map to manipulate.
+    deidx : bool, optional
+        Strip suffixes from subject IDs.
+    demux : bool, optional
+        Whether perform demultiplexing.
+    samples : list of str, optional
+        Sample ID list to include.
+    files : list or dict
+        Map of filepaths to sample IDs.
+    fp : str
+        Path to current alignment file.
+
+    Returns
+    -------
+    dict
+        Reshaped read map.
+    """
+    # strip indices from subjects
+    if deidx:
+        strip_index(rmap)
+
+    # demultiplex into multiple samples
+    if demux:
+        return demultiplex(rmap, samples)
+
+    # sample Id from filename
+    else:
+        return {files[fp]: rmap}
+
+
+def assign_rmap(rmap:     dict,
```

Comment:
```suggestion
def assign_readmap(readmap:     dict,
```
and corresponding conde changes

Reply:
I think replacing `rmap` with `readmap` is a good idea! And I changed them all. Meanwhile I am not sure if `rmap` as a variable is bad in the function, since abbreviation is common (for examples: `fh` and `res`) when there is no ambiguity.